### ISSUE TITLE
Endless API call

### ISSUE THREAD

**Issue by wanickols01** [Ananya2001-an/githubwars#22]
**Describe the bug**
The API call for 100 users is called continously upon opening the battling page (either). This causes a huge number of fetches and resources if it continues. 

**Expected behavior**
I would expect only calling this once. If more than 100 are needed we could simply increase the hard coded value in the code for this, but endlessly calling this is not preferred. It can call it once on page construction. 

**Screenshots**
![image](https://user-images.githubusercontent.com/78195616/218965615-1b405cb3-8d43-4e58-b42c-af49ffa1162f.png)

**Comment by Ananya2001-an** [comment]
@wanickols01 Yes you are right the api is getting called endlessly because I thought of getting all the users and provide them as a context to the children components but definitely it's not good in terms of performance. You can work on this if you want. Maybe reducing to a limited no of users let's say 1000 or something.

**Comment by Ananya2001-an** [comment]
Hey @wanickols01 are you still working on this?
